src/object/srv_ec_aggregate.c

```diff
@@ -2202,11 +2213,12 @@ agg_dkey(daos_handle_t ih, vos_iter_entry_t *entry,
 			DP_UOID(agg_entry->ae_oid), DP_KEY(&agg_entry->ae_dkey),
 			agg_entry->ae_is_leader ? "yes" : "no");
 		agg_reset_dkey_entry(&agg_param->ap_agg_entry, entry);
+		rc = agg_get_obj_handle(agg_entry, true);
```

Comment:
Minor: looks moving the peer location setting code in agg_get_obj_handle() to a separate function and call it here would be more clear.

Reply:
agreed,  how about make another cleanup patch, this PR only fix functional logic and make it fix the problem as fast as possible.

Reply:
sure